gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/ConcatenatedGigawordDocuments.java

```diff
@@ -18,18 +26,112 @@
 
 public class ConcatenatedGigawordDocuments implements Iterable<ConcatenatedGigawordDocuments.Article> {
   private final String concatenatedFileText;
+  private List<Article> articleList;
+  boolean annotated;
 
   private ConcatenatedGigawordDocuments(String concatenatedFileText) {
     this.concatenatedFileText = Objects.requireNonNull(concatenatedFileText);
+    this.annotated = false;
   }
 
+  private ConcatenatedGigawordDocuments(String concatenatedFileText, List<Article> aArticleList) {
+    this.concatenatedFileText = Objects.requireNonNull(concatenatedFileText);
+    this.articleList = aArticleList;
+    this.annotated = true;
+  }
+  
   public static ConcatenatedGigawordDocuments fromGigwordGZippedFile(Path p) throws IOException {
     return new ConcatenatedGigawordDocuments(GZIPByteSource.fromCompressed(p.toFile())
             .asCharSource(StandardCharsets.UTF_8).read());
   }
 
+  public static ConcatenatedGigawordDocuments fromAnnotatedGigwordGZippedFile(Path p) throws IOException {
+    // Initialize Streams
+    try (InputStream fileInputStream = new FileInputStream(p.toString());
+         GZIPInputStream gzipInputStream = new GZIPInputStream(fileInputStream);
+         InputStreamReader inputStreamReader = new InputStreamReader(gzipInputStream);
+         BufferedReader br = new BufferedReader(inputStreamReader)) {
+
+        String sCurrentLine;
+    
+        // variables for start of a new document
+        Pattern GIGAWORD_DOC_ELEMENT_PATTERN = Pattern.compile("<DOC id=\"(.*?)\".*>");
+        String currentDocId = "";
+    
+        // variables for saving articles
+        boolean headlineStarted = false;
+        boolean textStarted = false;
+        String docText = "";
+        List<Article> articleList = new ArrayList<>();
+    
+    
+        // read file
+        while ((sCurrentLine = br.readLine()) != null) {
+    
+            sCurrentLine = sCurrentLine.trim();
+    
+            // extract document ID
+            if (sCurrentLine.startsWith("<DOC id=")) {
+                Matcher m = GIGAWORD_DOC_ELEMENT_PATTERN.matcher(sCurrentLine);
+                if (m.find()) {
+                    currentDocId = m.group(1);
+                }
+                else {
+                    throw new RuntimeException("Missing document ID on article");
+                }
+            }
+    
+            // create article when end of document (</TEXT>) is reached
+            if (sCurrentLine.equals("</TEXT>")) {
+                articleList.add(new Article(currentDocId, docText));
```

Comment:
Should probably trim the docText here to remove any trailing spaces?

Reply:
Fixed in newest commit.